src/main/java/org/skriptlang/skript/bukkit/recipes/RecipeWrapper.java

```diff
@@ -0,0 +1,385 @@
+package org.skriptlang.skript.bukkit.recipes;
+
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.*;
+import org.bukkit.inventory.recipe.CookingBookCategory;
+import org.bukkit.inventory.recipe.CraftingBookCategory;
+import org.skriptlang.skript.bukkit.recipes.RecipeUtils.RecipeType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Wrapper class allowing the creation of a Recipe without being on initialization.
+ */
+public abstract class RecipeWrapper implements Recipe {
+
+	private final RecipeType recipeType;
+	private final NamespacedKey key;
+	private ItemStack result = null;
+	private List<String> errors = new ArrayList<>();
+
+	public RecipeWrapper(NamespacedKey key, RecipeType recipeType) {
+		this.key = key;
+		this.recipeType = recipeType;
+	}
+
+	public void setResult(ItemStack result) {
+		this.result = result;
+	}
+
+	@Override
+	public ItemStack getResult() {
+		return result;
+	}
+
+	public NamespacedKey getKey() {
+		return this.key;
+	}
+
+	public RecipeType getRecipeType() {
+		return recipeType;
+	}
+
+	public void addError(String error) {
+		errors.add(error);
+	}
+
+	public List<String> getErrors() {
+		return errors;
+	}
+
+	public abstract Recipe create();
+
+    public static class CraftingRecipeWrapper extends RecipeWrapper {
```

Comment:
this should be abstract if you aren't meant to make one directly (also what's up with the indentation?)